refactor: Rearrange registered events dictionaries.

Rohitth007 opened the pull request:
This refactor moves the dictionary `fetch_types` to `__init__`.
This dictionary which was previously in `_register_desired_events`
made it look like those were the events being registered to
while actually they were initial fetch events.

The function `_update_initial_data()` has been given a more
apt name `_fetch_initial_data()`.

Rohitth007 commented:
I tried to but couldn't satisfy PEP8's line length condition :(
83 > 79

neiljp commented on zulipterminal/model.py:
```diff
@@ -88,25 +88,42 @@ def __init__(self, controller: Any) -> None:
+        ]
+
+        # Events desired while in runtime.
+        self.desired_events_action: 'OrderedDict[str, Callable[[Event], None]]' = (
```
I don't think we need to modify this variable at all - and that will keep the main movement of the list into `__init__` a lot clearer.

Rohitth007 replied:
Oh ok, I'll reset this back to the original name.

neiljp reviewed (commented):
@Rohitth007 Thanks for the refactoring contribution! :+1: I'm glad you figured out the difference between the sets of strings and this should help others too :)

If you plan to contribute further I'd suggest checking the ZT code style guidelines or reviewing recent git log entries for examples.

zee-bit commented on zulipterminal/model.py:
```diff
@@ -96,17 +113,17 @@ def __init__(self, controller: Any) -> None:
                 ('update_message_flags',
-                 self._handle_update_message_flags_event),
+                    self._handle_update_message_flags_event),
                 ('update_display_settings',
-                 self._handle_update_display_settings_event),
+                    self._handle_update_display_settings_event),
```
I guess these diffs aren't needed. :)

Rohitth007 replied:
Don't remember messing with that, I'll remove them

zee-bit reviewed (commented):
This is a nice refactoring. :+1:
There are few points to mention, though;
1. You can squash the second commit, or drop it and amend the first commit since it just reverts the changes of your first commit. Also, we probably don't want to have an intermediate phase where the tests were broken.
2. As Neil already mentioned, we follow a specific commit style that you can have a look at in your `git log`, and it'll be convenient if you can adapt your commit to follow those guidelines. :)

Rohitth007 commented:
Sorry, I didn't get the 2nd point, was my code style wrong or my commit style? @zee-bit @neiljp

Rohitth007 commented:
I have reverted unnecessary changes and squashed commits as needed. This is ready to be merged, please take a look.Thank you for your suggestions. @zee-bit @neiljp

neiljp commented:
@Rohitth007 Thanks again for the refactor! I made a few minor textual adjustments and merged it manually as f9d09825896e54d49a1b89c10d6d098c9182ea0b :tada:
